gcloud/logging/_gax.py

```diff
@@ -467,17 +457,20 @@ def _value_pb_to_value(value_pb):
 
 
 def _struct_pb_to_mapping(struct_pb):
-    """Helper for :func:`_log_entry_pb_to_mapping`."""
+    """Helper for :func:`_log_entry_pb_to_mapping`.
+
+    Performs "impedance matching" between the protobuf attrs and the keys
+    expected in the JSON API.
+    """
     return dict([(key, _value_pb_to_value(struct_pb.fields[key]))
                  for key in struct_pb.fields])
 
 
 def _log_entry_pb_to_mapping(entry_pb):
     """Helper for :meth:`list_entries`, et aliae
 
-    Ideally, would use a function from :mod:`protobuf.json_format`, but
-    the right one isn't public.  See:
-    https://github.com/google/protobuf/issues/1351
```

Comment:
Yup, the `json_format` bit wouldn't help much, as we'd still have to map all the `with_undersoce` name to camelCase for equivalence with the JSON API.